backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/components/CassandraTypeTest.java

```diff
@@ -0,0 +1,107 @@
+ ****************************************************************/
+package org.apache.james.backends.cassandra.components;
+
+import static org.apache.james.backends.cassandra.components.CassandraType.InitializationStatus.ALREADY_DONE;
+import static org.apache.james.backends.cassandra.components.CassandraType.InitializationStatus.PARTIAL;
+import static org.apache.james.backends.cassandra.components.CassandraType.InitializationStatus.FULL;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.UserType;
+import com.datastax.driver.core.schemabuilder.CreateType;
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
+import org.apache.james.backends.cassandra.components.CassandraType.InitializationStatus;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+
+class CassandraTypeTest {
+    private static final String NAME = "typeName";
+    private static final CreateType STATEMENT = SchemaBuilder.createType(NAME);
+    private static final CassandraType TYPE = new CassandraType(NAME, STATEMENT);
+
+    @Test
+    void shouldRespectBeanContract() {
+        EqualsVerifier.forClass(CassandraType.class)
+                .withPrefabValues(CreateType.class, SchemaBuilder.createType("name1"), SchemaBuilder.createType("name2"))
+                .verify();
+    }
+
+    @Test
+    void initializeShouldExecuteCreateStatementAndReturnFullWhenTypeDoesNotExist() {
+        KeyspaceMetadata keyspace = mock(KeyspaceMetadata.class);
+        when(keyspace.getUserType(NAME)).thenReturn(null);
+        Session session = mock(Session.class);
+
+        assertThat(TYPE.initialize(keyspace, session))
+                .isEqualByComparingTo(FULL);
+
+        verify(keyspace).getUserType(NAME);
+        verify(session).execute(STATEMENT);
+    }
+
+    @Test
+    void initializeShouldExecuteReturnAlreadyDoneWhenTypeExists() {
+        KeyspaceMetadata keyspace = mock(KeyspaceMetadata.class);
+        when(keyspace.getUserType(NAME)).thenReturn(mock(UserType.class));
+        Session session = mock(Session.class);
+
+        assertThat(TYPE.initialize(keyspace, session))
+                .isEqualByComparingTo(ALREADY_DONE);
+
+        verify(keyspace).getUserType(NAME);
+        verify(session, never()).execute(STATEMENT);
+    }
+
+    @Nested
+    static class InitializationStatusTest {
+        @ParameterizedTest
+        @MethodSource("combinationReduceProvider")
```

Comment:
AFAIK, you can have a @ParameterizedTest without having an ugly method-as-string
You should grep James codebase

Reply:
Sure, but you should give the same name to your provider.

Reply:
that way you don't have to use a string and you will be refactoring proof

Reply:
IntelliJ flags the MethodSource as unused now.